linera-execution/src/system.rs

```diff
@@ -517,9 +517,31 @@ where
                 for (chain_id, stream_id, next_index) in streams {
                     let subscriptions = self
                         .event_subscriptions
-                        .get_mut_or_default(&(chain_id, stream_id))
+                        .get_mut_or_default(&(chain_id, stream_id.clone()))
+                        .await?;
+                    if subscriptions.next_index >= next_index {
+                        continue;
+                    }
+                    subscriptions.next_index = next_index;
```

Comment:
Maybe we should update the index after we act on it.

Reply:
I moved this line to after we put the information into the transaction tracker.

Or do you mean we should update it only after calling `process_streams` in all the applications? That would complicate the code quite a bit, because we use the same code to call `process_streams`, regardless of whether we just processed `UpdateStreams` or any other operation or message that may have added a new subscription.